metals/src/main/scala/scala/meta/internal/implementation/SuperMethodProvider.scala

```diff
@@ -0,0 +1,285 @@
+      findSymbol: String => Option[SymbolInformation]
+  ): Option[List[SymbolInformation]] = {
+    val classSymbolInformationMaybe =
+      findClassInfo(
+        msi.symbol,
+        msi.symbol.owner,
+        documentWithPath.textDocument,
+        findSymbol
+      )
+    val methodInfo = msi.signature.asInstanceOf[MethodSignature]
+
+    classSymbolInformationMaybe.map(classSymbolInformation => {
+      calculateClassSuperHierarchy(classSymbolInformation, findSymbol)
+        .flatMap(matchMethodInClass(_, msi, methodInfo, findSymbol))
+    })
+  }
+
+  private def matchMethodInClass(
+      superClass: SymbolWithAsSeenFrom,
+      baseMethodSymbolInformation: SymbolInformation,
+      baseMethodInfo: MethodSignature,
+      findSymbol: String => Option[SymbolInformation]
+  ): Option[SymbolInformation] = {
+    superClass.symbolInformation.signature match {
+      case classSig: ClassSignature =>
+        classSig.getDeclarations.symlinks
+          .map(methodSlink =>
+            for {
+              mSymbolInformation <- findSymbol(methodSlink)
+              if mSymbolInformation.isMethod
+              methodSignature = mSymbolInformation.signature
+                .asInstanceOf[MethodSignature]
+              if checkSignaturesEqual(
+                mSymbolInformation,
+                methodSignature,
+                baseMethodSymbolInformation,
+                baseMethodInfo,
+                superClass.asSeenFrom,
+                findSymbol
+              )
+            } yield mSymbolInformation
+          )
+          .collectFirst { case Some(value) => value }
+      case _ =>
+        None
+    }
+
+  }
+
+  private def getSuperClasses(
+      symbolInformation: SymbolInformation,
+      findSymbol: String => Option[SymbolInformation],
+      skip: scala.collection.mutable.Set[SymbolInformation],
+      asSeenFrom: Map[String, String]
+  ): List[SymbolWithAsSeenFrom] = {
+    if (skip.exists(_.symbol == symbolInformation.symbol)) {
+      List.empty
+    } else {
+      skip += symbolInformation
+      symbolInformation.signature match {
+        case classSignature: ClassSignature =>
+          val parents = classSignature.parents
+            .collect { case x: TypeRef => x }
+            .filterNot(typeRef => stopSymbols.contains(typeRef.symbol))
+            .filterNot(typeRef => skip.exists(_.symbol == typeRef.symbol))
+          val parentsHierarchy = parents
+            .flatMap(p => findSymbol(p.symbol).map((_, p)))
+            .map {
+              case (si, p) =>
+                val parentASF = AsSeenFrom.calculateAsSeenFrom(
+                  p,
+                  classSignature.typeParameters
+                )
+                val currentASF =
+                  AsSeenFrom.translateAsSeenFrom(asSeenFrom, parentASF)
+                getSuperClasses(si, findSymbol, skip, currentASF)
+            }
+            .toList
+            .reverse
+            .flatten
+          val outASF =
```

Comment:
Do we need to do it here? `toRealNames` should really be translated at the point we are comparing two methods from two specific classes.